agent/peering_endpoint.go

```diff
@@ -18,30 +20,29 @@ func (s *HTTPHandlers) PeeringEndpoint(resp http.ResponseWriter, req *http.Reque
 		return nil, HTTPError{StatusCode: http.StatusBadRequest, Reason: "Must specify a name to fetch."}
 	}
 
-	entMeta := s.agent.AgentEnterpriseMeta()
-	if err := s.parseEntMetaPartition(req, entMeta); err != nil {
-		return nil, err
-	}
-
 	// Switch on the method
 	switch req.Method {
 	case "GET":
-		return s.peeringRead(resp, req, name, entMeta.PartitionOrEmpty())
+		return s.peeringRead(resp, req, name)
 	case "DELETE":
-		return s.peeringDelete(resp, req, name, entMeta.PartitionOrEmpty())
+		return s.peeringDelete(resp, req, name)
 	default:
 		return nil, MethodNotAllowedError{req.Method, []string{"GET", "DELETE"}}
 	}
 }
 
 // peeringRead fetches a peering that matches the name and partition.
 // This assumes that the name and partition parameters are valid
-func (s *HTTPHandlers) peeringRead(resp http.ResponseWriter, req *http.Request, name, partition string) (interface{}, error) {
+func (s *HTTPHandlers) peeringRead(resp http.ResponseWriter, req *http.Request, name string) (interface{}, error) {
 	args := pbpeering.PeeringReadRequest{
 		Name:       name,
 		Datacenter: s.agent.config.Datacenter,
-		Partition:  partition, // should be "" in OSS
 	}
+	var entMeta acl.EnterpriseMeta
```

Comment:
just out of curiosity, what were the trade offs for deciding to move this check into `peeringRead` and `peeringWrite`?  As opposed to keeping it above,  in the general `PeeringEndpoint` func?

Reply:
Consistency mostly. Almost all other endpoints follow this sort of structure for extracting partition/namespace data from requests to populate rpc fields.